source-pendo/source_pendo/resources.py

```diff
@@ -0,0 +1,181 @@
+        binding_index: int,
+        state: ResourceState,
+        task: Task,
+        all_bindings
+    ):
+        common.open_binding(
+            binding,
+            binding_index,
+            state,
+            task,
+            fetch_snapshot=functools.partial(
+                snapshot,
+                http,
+                entity,
+            ),
+            tombstone=Resource(_meta=Resource.Meta(op="d"))
+        )
+
+    resources = [
+        common.Resource(
+            name=resource_name,
+            key=["/id"],
+            model=Resource,
+            open=functools.partial(open, entity),
+            initial_state=ResourceState(),
+            initial_config=ResourceConfig(
+                name=resource_name, interval=timedelta(seconds=30)
+            ),
+            schema_inference=True,
+        )
+        for (entity, resource_name) in RESOURCE_TYPES
+    ]
+
+    return resources
+
+def metadata(
+        log: Logger, http: HTTPMixin, config: EndpointConfig
+) -> list[common.Resource]:
+
+    async def snapshot(http: HTTPMixin, entity: str, log: Logger) -> AsyncGenerator[Metadata, None]:
+        metadata = await fetch_metadata(http=http, entity=entity, log=log)
+        for m in metadata:
+            yield m
+
+    def open(
+        entity: str,
+        binding: CaptureBinding[ResourceConfig],
+        binding_index: int,
+        state: ResourceState,
+        task: Task,
+        all_bindings
+    ):
+        common.open_binding(
+            binding,
+            binding_index,
+            state,
+            task,
+            fetch_snapshot=functools.partial(
+                snapshot,
+                http,
+                entity,
+            ),
+            tombstone=Metadata(_meta=Metadata.Meta(op="d"))
+        )
+
+    metadata = [
+        common.Resource(
+            name=resource_name,
+            key=["/_meta/row_id"],
+            model=Metadata,
+            open=functools.partial(open, entity),
+            initial_state=ResourceState(),
+            initial_config=ResourceConfig(
+                name=resource_name, interval=timedelta(seconds=30)
+            ),
+            schema_inference=True,
+        )
+        for (entity, resource_name) in METADATA_TYPES
+    ]
+
+    return metadata
+
+def events(
+        log: Logger, http: HTTPMixin, config: EndpointConfig
+) -> list[common.Resource]:
+    def open(
+        entity: str,
+        timestamp_field: str,
+        binding: CaptureBinding[ResourceConfig],
+        binding_index: int,
+        state: ResourceState,
+        task: Task,
+        all_bindings
+    ):
+        common.open_binding(
+            binding,
+            binding_index,
+            state,
+            task,
+            fetch_changes=functools.partial(
+                fetch_events,
+                http,
+                entity,
+                timestamp_field
+            )
+        )
+
+    start_date = config.startDate
```

Comment:
Usually we would allow an empty start date configuration, and have some default like start capturing X days ago. Would that make sense to do here?

Reply:
Yep, I think that makes sense. I actually used Pydantic's [`default_factory`](https://docs.pydantic.dev/latest/concepts/fields/#default-values) this morning to do just that, but the default value wasn't populated into the config. I think I could have something like `config.startDate = config.startDate` in the `all_resources`  method to make sure that default value get populated. I'll keep looking into it.